ext/phar/phar_object.c

```diff
@@ -482,7 +482,7 @@ PHP_METHOD(Phar, mount)
 carry_on:
 		if (SUCCESS != phar_mount_entry(pphar, actual, actual_len, path, path_len)) {
 			zend_throw_exception_ex(phar_ce_PharException, 0, "Mounting of %s to %s within phar %s failed", path, actual, arch);
-			if (path && path == entry) {
+			if (entry && path == entry) {
```

Comment:
What's the simplification here?

Reply:
unified with (simplified) cond a few lines below

Reply:
It can likely be further simplified to `path == entry` then. See: https://github.com/phpinternalsbook/PHP-Internals-Book/issues/139

Reply:
I am not convinced freeing twice is the cleanest desing. So I prefer to keep the cond here...

Reply:
See my response over in the linked issue.